### PR TITLE
Add support for decorated class expression

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Fixes #92

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -302,33 +305,62 @@ <h1>Runtime Semantics: ClassElementEvaluation</h1>
 
   </emu-clause>
 
+  <!-- es6num="14.5.15" -->
+  <emu-clause id="sec-runtime-semantics-bindingclassdeclarationevaluation">
+    <h1>Runtime Semantics: BindingClassDeclarationEvaluation</h1>
+    <emu-grammar>ClassDeclaration : <ins>DecoratorList?</ins> `class` BindingIdentifier ClassTail</emu-grammar>
+    <emu-alg>
+      1. <ins>If |DecoratorList_opt| is present, let _decorators_ be the result of performing DecoratorListEvaluation of |DecoratorList|.</ins>
```
Comment (nicolo-ribaudo) [inline spec.html:323]: All this section has just been unindented, as you can see from https://github.com/tc39/proposal-decorators/pull/93/files?w=1. The only change is here (`DecoratorList` -> `DecoratorList_opt`)

**Review by littledan — APPROVED** [review]
This patch looks good, but how worried are you about the way that it enables decorators to come either before or after `export default`?

**Comment by littledan** [comment]
Based on the discussion in #92, seems like this patch should require parens in the confusing case `export default @foo class C { }`. I think this would be a matter of adding one more item to the long negative lookahead list in `export default AssignmentExpression` cases, in https://tc39.github.io/ecma262/#prod-ExportDeclaration . All the other function/class declaration forms have the negative lookahead, so it makes perfect sense to have one for `@` as well. (We probably actually need that negative lookahead even without this patch, to avoid ambiguity.) @nicolo-ribaudo would you be interested in making this revision?

**Comment by ljharb** [comment]
Why is that case confusing? Requiring parens there seems very unfortunate.

**Comment by littledan** [comment]
@ljharb Parens aren't required, it's just that without parens you place the decorators before `export`. You also need parentheses to get at these semantics:

```js
export default (function f() { f() });
```

That's a recursive anonymous function--it doesn't export the binding `f` outside of the function itself, the way it would if the parens weren't there.

**Comment by rbuckton** [comment]
Parens are already required to distinguish between a class declaration and a class expression in an ExportDeclaration.

**Review by littledan — APPROVED** [review]

**Pull request merged**